XenAdmin/MainWindow.cs

```diff
@@ -716,73 +711,80 @@ private void ShowAboutDialogOnStartup()
+                        break;
+                    case "license":
+                        log.DebugFormat("CLI: Installing license from {0}", args[1]);
+                        LaunchLicensePicker(args[1]);
+                        break;
+                    case "restore":
+                        log.DebugFormat("CLI: Restoring host backup from {0}", args[1]);
+                        new RestoreHostFromBackupCommand(this, null, args[1]).Run();
+                        break;
+                    case "search":
+                        log.DebugFormat("CLI: Importing saved XenSearch from '{0}'", args[1]);
+                        new ImportSearchCommand(this, args[1]).Run();
+                        break;
+                    case "connect":
+                        log.DebugFormat("CLI: Connecting to server '{0}'", args[1]);
 
-                    var connection = new XenConnection
-                    {
-                        Hostname = args[0],
-                        Port = ConnectionsManager.DEFAULT_XEN_PORT,
-                        Username = args.Length > 1 ? args[1] : "",
-                        Password = args.Length > 2 ? args[2] : ""
-                    };
+                        var connection = new XenConnection
+                        {
+                            Hostname = args[1],
+                            Port = ConnectionsManager.DEFAULT_XEN_PORT,
+                            Username = args.Length > 2 ? args[2] : "",
+                            Password = args.Length > 3 ? args[3] : ""
+                        };
 
-                    if (ConnectionsManager.XenConnectionsContains(connection))
-                        break;
+                        if (ConnectionsManager.XenConnectionsContains(connection))
+                            break;
 
-                    lock (ConnectionsManager.ConnectionsLock)
-                        ConnectionsManager.XenConnections.Add(connection);
+                        lock (ConnectionsManager.ConnectionsLock)
+                            ConnectionsManager.XenConnections.Add(connection);
 
-                    XenConnectionUI.BeginConnect(connection, true, null, false);
-                    break;
-                case ArgType.None:
-                    if (Launched)
-                    {
-                        // The user has launched the splash screen, but we're already running.
-                        // Draw their attention.
-                        HelpersGUI.BringFormToFront(this);
-                        Activate();
-                    }
-                    break;
-                case ArgType.Passwords:
-                    Trace.Assert(false);
-                    break;
+                        XenConnectionUI.BeginConnect(connection, true, null, false);
+                        break;
+                    default:
+                        log.Warn("CLI: Wrong syntax or unknown command line options.");

```

Comment:
We can't do this for security reasons: it may be a highly improbable case, but if a user passed in `password connect server username` instead of `connect server username password`, the password would end up showing in the logs. This is the reason why logging around the CLI commands is sparse in general.

Reply:
oh right! makes sense 👍